### ISSUE TITLE
Duplicated port attribute on db_instance data source 

### ISSUE THREAD

**Issue by gruzewski** [hashicorp/terraform-provider-aws#31441]
### Description

Hello,

I am trying to understand differences between two potentially the same attributes on [Data Source: aws_db_instance](https://registry.terraform.io/providers/hashicorp/aws/latest/docs/data-sources/db_instance). At the moment, we have:

> [db_instance_port](https://registry.terraform.io/providers/hashicorp/aws/latest/docs/data-sources/db_instance#db_instance_port) - Port that the DB instance listens on.

and

> [port](https://registry.terraform.io/providers/hashicorp/aws/latest/docs/data-sources/db_instance#port) - Database port.

The former was the first one, with the latter [added in 2017](https://github.com/hashicorp/terraform-provider-aws/commit/fab437e8bf755c924889b28d0cabff809afe4d4a). I tried searching any information about differences between those two. Could you throw light on why we have those two attributes? If they are the same, let's either make a note in docs or deprecate one of them.

### Would you like to implement a fix?

None

**Comment by gruzewski** [comment]
Thanks, @mattburgess for a speedy reply and links :)

TL;DR: `port` refers to Aurora's endpoint (which is a cluster of DB instances with some kind of proxy in front) and that can be different than a port of a db instance. If you're not using Aurora, you should probably use `db_instance_port`.

Based on [instance_data_source.go#L276-L288](https://github.com/hashicorp/terraform-provider-aws/blob/main/internal/service/rds/instance_data_source.go#L276-L288) looks like `port` can be empty.

```go
	// Per AWS SDK Go docs:
	// The endpoint might not be shown for instances whose status is creating.
	if dbEndpoint := v.Endpoint; dbEndpoint != nil {
		d.Set("address", dbEndpoint.Address)
		d.Set("endpoint", fmt.Sprintf("%s:%d", aws.StringValue(dbEndpoint.Address), aws.Int64Value(dbEndpoint.Port)))
		d.Set("hosted_zone_id", dbEndpoint.HostedZoneId)
		d.Set("port", dbEndpoint.Port)
	} else {
		d.Set("address", nil)
		d.Set("endpoint", nil)
		d.Set("hosted_zone_id", nil)
		d.Set("port", nil)
	}
```

That could happen if someone creates an Aurora DB cluster but doesn't add any Database instance as mentioned [here](https://repost.aws/knowledge-center/aurora-endpoints-creating-status). 

Also, [create-db-cluster](https://docs.aws.amazon.com/cli/latest/reference/rds/create-db-cluster.html) and [create-db-instance](https://docs.aws.amazon.com/cli/latest/reference/rds/create-db-instance.html) RDS CLI commands throw more light on why endpoint port could be different from the DB instance one.

> If you create an Aurora DB cluster, the request creates an empty cluster. You must explicitly create the writer instance for your DB cluster using the [CreateDBInstance](https://docs.aws.amazon.com/AmazonRDS/latest/APIReference/API_CreateDBInstance.html) operation

> The new DB instance can be an RDS DB instance, or it can be a DB instance in an Aurora DB cluster. For an Aurora DB cluster, you can call this operation multiple times to add more than one DB instance to the cluster.

Both of them take a port as a parameter :)

[This article](https://docs.aws.amazon.com/AmazonRDS/latest/AuroraUserGuide/Aurora.Overview.Endpoints.html) also explains a bit about why this could be happening.

> Amazon Aurora typically involves a cluster of DB instances instead of a single instance. Each connection is handled by a specific DB instance. When you connect to an Aurora cluster, the host name and port that you specify point to an intermediate handler called an endpoint. Aurora uses the endpoint mechanism to abstract these connections.

The mystery is solved! :) But it would be good to update the doc. I can give it a go and draft a PR.

**Comment by gruzewski** [comment]
@AndrewCharlesHay, thanks for following up. I have just created one here -> #31831
